src/core_atmosphere/diagnostics/Registry_isobaric.xml

```diff
@@ -161,18 +177,137 @@
+             description="Cloud water mixing ratio vertically interpolated to isobaric surfaces defined in qc_iso_levels"/>
+
+        <var name="uzonal_isobaric"  type="real" dimensions="nIsoLevelsUZONAL nCells Time" units="m s^{-1}"
+             description="Zonal component of reconstructed horizontal velocity at cell centers vertically interpolated to isobaric surfaces defined in uzonal_iso_levels"/>
+
+        <var name="umeridional_isobaric"  type="real" dimensions="nIsoLevelsUMERIDIONAL nCells Time" units="m s^{-1}"
+             description="Meridional component of reconstructed horizontal velocity at cell centers vertically interpolated to isobaric surfaces defined in umeridional_iso_levels"/>
+
+        <var name="w_isobaric"       type="real" dimensions="nIsoLevelsW nCells Time" units="m s^{-1}"
+             description="Vertical velocity vertically interpolated to isobaric surfaces defined in w_iso_levels"/>
 
         <var name="meanT_500_300"    type="real" dimensions="nCells Time" units="K"
              description="Mean temperature in the 300 hPa - 500 hPa layer"/>
 
-        <var name="t_iso_levels" type="real" dimensions="nIsoLevelsT" units="Pa"
+        <var name="t_iso_levels"     type="real" dimensions="nIsoLevelsT" units="Pa"
              description="Levels for vertical interpolation of temperature to isobaric surfaces"/>
 
-        <var name="z_iso_levels" type="real" dimensions="nIsoLevelsZ" units="Pa"
+        <var name="z_iso_levels"     type="real" dimensions="nIsoLevelsZ" units="Pa"
              description="Levels for vertical interpolation of height to isobaric surfaces"/>
 
+        <var name="relhum_iso_levels" type="real" dimensions="nIsoLevelsRELHUM" units="Pa"
+             description="Levels for vertical interpolation of relative humidity to isobaric surfaces"/>
+
+        <var name="qc_iso_levels"    type="real" dimensions="nIsoLevelsQC" units="Pa"
+             description="Levels for vertical interpolation of cloud water mixing ratio to isobaric surfaces"/>
+
+        <var name="uzonal_iso_levels" type="real" dimensions="nIsoLevelsUZONAL" units="Pa"
+             description="Levels for vertical interpolation of zonal component of reconstructed horizontal velocity at cell centers to isobaric surfaces"/>
+
+        <var name="umeridional_iso_levels" type="real" dimensions="nIsoLevelsUMERIDIONAL" units="Pa"
+             description="Levels for vertical interpolation of meridional component of reconstructed horizontal velocity at cell centers to isobaric surfaces"/>
+
+        <var name="w_iso_levels"     type="real" dimensions="nIsoLevelsW" units="Pa"
+             description="Levels for vertical interpolation of vertical velocity to isobaric surfaces"/>
+
 </var_struct>
+
+<nml_record name="isobaric_diagnostics" in_defaults="true">
+
+        <nml_option name="config_isobaric_diagnostic_num_levels_t" type="integer" default_value="5" in_defaults="false"
+             units="-"
```

Comment:
The remaining wart is a requirement that the number of items in csv string config_isobaric_diagnostic_levels_X must match the size specified in config_isobaric_diagnostics_num_levels_X. This is checked at runtime after parsing the csv. Open to good ideas for DRYing this.